pkg/client/cli/extensions/builtin.go

```diff
@@ -48,6 +48,15 @@ func builtinExtensions(ctx context.Context) map[string]ExtensionInfo {
 								`If this flag is given multiple times, then it will only intercept traffic that matches *all* of the specifiers. ` +
 								`(default "auto" if you are logged in with 'telepresence login', default "all" otherwise)`,
 						},
+						"protocol": {
+							Type: "string",
+							Usage: `` +
+								`Supported protocols are ` +
+								`"http" (Plaintext HTTP/1.1), ` +
+								`"http2" (Plaintext HTTP/2), ` +
+								`"tcp" (Opaque TCP stream, implies --http_match=all), and ` +
```

Comment:
it feels a little strange that `--http-protocol=tcp` is a valid value. Maybe `--net-protocol` would make more sense?
At the same time i understand that this makes it consistent with the other `--http-*` flags so it does work from that perspective.

Reply:
Yeah, "tcp" was more meant to signify "opaque", i.e. to tell Envoy to just dispatch without any attempt to do L7 routing. The name stems from Istio's port selection. I think I'll just remove it for now.